conans/client/installer.py

```diff
@@ -563,6 +563,8 @@ def _call_package_info(self, conanfile, package_folder, ref):
                     if conanfile._conan_dep_cpp_info is None:
                         try:
                             conanfile.cpp_info._raise_if_mixing_components()
+                            conanfile.cpp_info._raise_if_not_scoped_requires_in_components()
```

Comment:
This is a bit ugly, 3 different calls to private methods. Maybe could be grouped in one call?